components/page/DevPageProjects.tsx

```diff
@@ -0,0 +1,194 @@
+import styles from '@components/page/devpage.module.css'
+
+import {
+    Card,
+    CardHeader,
+    CardTitle,
+    CardBody,
+    CardExpandableContent,
+    Button,
+  } from '@patternfly/react-core'
+import { useState } from 'react'
+
+/*
+    component for starter projects dropdown 
+*/
+
+//TODO: subdirectories
+//TODO: fix scroll bar
+//TODO: Show less Tags
+
+/**
+ * Representation of a Git source for  a starter project
+ *
+ * @param checkoutFrom - object with optional remote and revision keys 
+ *                       and specifies where project should be checked out
+ * @param remote - remote name in remotes that should be used as the init and download url
+ * @param revision - revision to checkout from
+ * @param remotes - remotes map; should be init. to git project
+ */
+interface Git {
+    checkoutFrom?:{
+        remote?:string,
+        revision?:string
+    },
+    remotes:Record<string,string>
+}
+
+/**
+ * Representation of a single starter project
+ * 
+ * @remarks
+ *  For more details, check out the devfile API reference at https://docs.devfile.io/devfile/2.1.0/user-guide/api-reference.html
+ * 
+ * @param name - name of starter project
+ * @param description: description of starter project
+ * @param git - git source of starter project
+ * @param zip - zip source of start project with key location
+ * @param location - zip project's source location address
+ * @param subDir - subdirectory of zip to be used as the root for the starter project
+ */
+interface Project { 
+    name:string,
+    description?:string,
+    git?:Git,
+    zip?:{
+        location:string
+    },
+    subDir?:string
+}
+
+/**
+ * props for start projects dropdown component
+ * 
+ * @param starterProjects list of starter projects
+ */
+interface Props { 
+    starterProjects: Project[]
+}
+
+/**
+ * extract zip url from project git source
+ * 
+ * @param git - git source of project, assumes github url of root directory of project
+ * 
+ * @throws TypeError 
+ *      thrown if git remotes isn't configured properly
+ */
+function getURLForGit(git:Git){
+    var url = ""
+    const keys = Object.keys(git.remotes)
+
+    if(keys.length===1){
+        url = git.remotes[keys[0]]
+
+    } else if(git.checkoutFrom&&git.checkoutFrom.remote) {//should always be true if keys.length!=1
+        url=git.remotes[git.checkoutFrom.remote]
+
+    } else {
+        throw  TypeError("Invalid git remotes")
+    }
+    url = url.slice(0,url.length-4).replace("github.com","api.github.com/repos") +"/zipball/" //remove ".git" from link and convert to api zip link
```

Comment:
neat! 👍 

We should check for `.git` in the link before just slicing off the last 4. https://github.com/odo-devfiles/nodejs-ex and https://github.com/odo-devfiles/nodejs-ex.git redirect to the same page

Reply:
resolved